### PR TITLE
Federation test util fix - 1

### PR DESCRIPTION
**Pull request by mwielgus**
* Make nil and empty map equivalent when testing ObjectMeta equivalence
* Add a function that waits for a particular content inside federated store

cc: @nikhiljindal @madhusudancs

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36605)

### CODE REVIEW

File: federation/pkg/federation-controller/util/test/test_helper.go [hunk 1]
```diff
@@ -280,3 +280,29 @@ func WaitForStoreUpdate(store util.FederatedReadOnlyStore, clusterName, key stri
...
+	checkFunction CheckingFunction) error {
+	retryInterval := 500 * time.Millisecond
+	var lastError error
+	err := wait.PollImmediate(retryInterval, timeout, func() (bool, error) {
+		item, found, err := store.GetByKey(clusterName, key)
+		if err != nil || !found {
+			return found, err
+		}
+		runtimeObj := item.(runtime.Object)
+		lastError = checkFunction(runtimeObj)
```
Comment (nikhiljindal) [inline federation/pkg/federation-controller/util/test/test_helper.go]: Add a vlog for lastError

File: federation/pkg/federation-controller/util/meta.go [hunk 2]
```diff
@@ -65,10 +65,10 @@ func ObjectMetaEquivalent(a, b api_v1.ObjectMeta) bool {
 	if a.Namespace != b.Namespace {
 		return false
 	}
-	if !reflect.DeepEqual(a.Labels, b.Labels) {
+	if !reflect.DeepEqual(a.Labels, b.Labels) && (len(a.Labels) != 0 || len(b.Labels) != 0) {
```
Comment (nikhiljindal) [inline federation/pkg/federation-controller/util/meta.go]: Same. will be good to add a test that ensures that objects with null and empty labels are considered equal.

**Comment by nikhiljindal** [comment]
LGTM with 2 suggestions.
Feel free to self apply lgtm label

File: federation/pkg/federation-controller/util/meta.go (hunk 2 above)
Comment (mwielgus) [inline federation/pkg/federation-controller/util/meta.go]: Done.

**Comment by mwielgus** [comment]
Comments applied. Adding LGTM.

**Pull request merged**